cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -272,11 +272,13 @@ class PyValue {
               return internal::InvalidValue(obj,
```

Comment:
Can you add a comment in the pandas timestamp case a few lines above about that case not needing to apply to subtract the offset, because pandas timestamps `value` is already nanoseconds in UTC (in contrast to datetime.datetime's microseconds)

Reply:
added.